ambry-router/src/test/java/com.github.ambry.router/InMemoryRouter.java

```diff
@@ -319,16 +327,21 @@ public void close() {
    * @param futureResult the {@link FutureResult} to update in case the operation has to be completed.
    * @param callback the {@link Callback} that needs to be invoked in case the operation has to be completed. Can be
    *                 null.
+   * @return if {@code true}, the rest of the code can continue.
    */
-  private void handlePrechecks(FutureResult futureResult, Callback callback) {
+  private boolean handlePrechecks(FutureResult futureResult, Callback callback) {
+    boolean continueOp = true;
     if (!routerOpen.get()) {
+      continueOp = false;
       completeOperation(futureResult, callback, null,
           new RouterException("Cannot accept operation because Router is closed", RouterErrorCode.RouterClosed));
     } else if (verifiableProperties.containsKey(OPERATION_THROW_EARLY_RUNTIME_EXCEPTION)) {
       throw new RuntimeException(OPERATION_THROW_EARLY_RUNTIME_EXCEPTION);
```

Comment:
No, because it throws immediately. That is an indication that op should not be continued. Also since this is an exception, the value of `continueOp` doesn't matter because it won't be returned